src/views/Timeline.vue

```diff
@@ -143,47 +123,69 @@ export default {
 	},
 
 	methods: {
-		async fetchContent() {
-			// only one simultaneous page load
-			if (this.loadingPage) {
-				return
+		async getContent(params) {
+			if (this.done) {
+				return Promise.resolve([])
 			}
 
 			// cancel any pending requests
 			if (this.cancelRequest) {
 				this.cancelRequest('Changed view')
 			}
 
-			// if we don't already have some cached data let's show a loader
-			if (this.timeline.length === 0) {
-				this.$emit('update:loading', true)
-			}
-			this.error = null
-			this.loadingPage = true
-
-			// init cancellable request
+			// done loading even with errors
 			const { request, cancel } = cancelableRequest(getPhotos)
 			this.cancelRequest = cancel
 
 			try {
-				// get content and current folder info
 				const files = await request(this.onlyFavorites, {
-					page: this.page,
-					perPage: this.gridConfig.count * 5, // we load 5 rows,
+					page: params.offset,
+					perPage: 30,
 				})
-				this.$store.dispatch('updateTimeline', files)
-				this.$store.dispatch('appendFiles', files)
 
-				// next time we load this script, we load the next page if the list returned
-				if (files.length === this.gridConfig.count * 5) {
-					this.page++
-				} else {
-					console.debug('We loaded the last page')
+				if (files.length === params.offset === 0) {
+					this.isEmpty = true
+				}
+
+				if (files.length !== 30) {
```

Comment:
Same here (grid config)

Reply:
Shall we make a gridconfig that is a setting?

Reply:
based on gridconfig now